server/src/main/java/io/crate/user/SecureHash.java

```diff
@@ -161,41 +161,40 @@ public static SecureHash fromXContent(XContentParser parser) throws IOException
         boolean hasPassword = false;
 
         while ((currentToken = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-            if (currentToken == XContentParser.Token.FIELD_NAME) {
-                while ((currentToken = parser.nextToken()) != XContentParser.Token.END_OBJECT) { // secure_hash
-                    hasPassword = true;
-                    if (currentToken == XContentParser.Token.FIELD_NAME) {
-                        String currentFieldName = parser.currentName();
-                        currentToken = parser.nextToken();
-                        switch (currentFieldName) {
-                            case X_CONTENT_KEY_ITERATIONS:
-                                if (currentToken != XContentParser.Token.VALUE_NUMBER) {
-                                    throw new ElasticsearchParseException(
-                                        "failed to parse SecureHash, 'iterations' value is not a number [{}]", currentToken);
-                                }
-                                iterations = parser.intValue();
-                                break;
-                            case X_CONTENT_KEY_HASH:
-                                if (currentToken.isValue() == false) {
-                                    throw new ElasticsearchParseException(
-                                        "failed to parse SecureHash, 'hash' does not contain any value [{}]", currentToken);
-                                }
-                                hash = parser.binaryValue();
-                                break;
-                            case X_CONTENT_KEY_SALT:
-                                if (currentToken.isValue() == false) {
-                                    throw new ElasticsearchParseException(
-                                        "failed to parse SecureHash, 'salt' does not contain any value [{}]", currentToken);
-                                }
-                                salt = parser.binaryValue();
-                                break;
-                            default:
-                                throw new ElasticsearchParseException("failed to parse secure_hash");
+            while (currentToken == XContentParser.Token.FIELD_NAME) {
```

Comment:
There was an issue here where the `iterations` was not read from XContent, and initially I thought to fix it independently, but since it doesn't affect login checks, I changed my mind, let me know though if you think otherwise.